src/core/execute.c

```diff
@@ -1451,7 +1451,8 @@ void exec_context_dump(const ExecContext *c, FILE* f, const char *prefix) {
         if (exec_context_restrict_namespaces_set(c)) {
                 _cleanup_free_ char *s = NULL;
 
-                r = namespace_flags_to_string(c->restrict_namespaces, &s);
+                /* Restrict all namespaces is zero, so in that case, we should print out all namespaces. */
+                r = namespace_flags_to_string(c->restrict_namespaces ?: NAMESPACE_FLAGS_ALL, &s);
```

Comment:
Hm? `RestrictNamespaces=yes` is mapped to zero, and that means no namespaces can be accessible from the service. So, here nothing should be printed, no??

Reply:
Yes, `RestrictNamespaces=yes` maps to `c->restrict-namespaces = 0`, so this ends up printing `RestrictNamespaces = n/a` below. For a naive user like myself, this seems to imply that RestrictNamespaces=no. I thought it would be more intuitive to print all namespaces instead.

With that said, this is a minor usability thing and maybe not worth the confusion. Let me revert the second part of this commit.